src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,53 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class LineDrawerAdapter implements Job2dDriver {
+
+    private DrawPanelController drawPanelController;
+    private ILine line;
+    private int startX = 0;
+    private int startY = 0;
+
+    public LineDrawerAdapter(DrawPanelController drawPanelController, LineTypeEnum lineTypeEnum) {
+
+        this.drawPanelController = drawPanelController;
+
+        switch (lineTypeEnum) {
+            case BASIC:
+                this.line = LineFactory.getBasicLine();
```

Comment:
Not bad, but, this is not a generic solution. (e.g. using other line types is not yet possible)
I understand the idea, but such direct coupling is unnecessary (KISS principle)